openpype/tools/sceneinventory/switch_dialog.py

```diff
@@ -1300,17 +1303,26 @@ def _trigger_switch(self, loader=None):
 
             try:
                 switch_container(container, repre_doc, loader)
-            except Exception:
+            except Exception as exc:
+                if isinstance(exc, (LoaderSwitchNotImplementedError,
```

Comment:
Maybe there should be one exception that is used as base for the others?

Reply:
So this code would be
```
try:
    switch_container(container, repre_doc, loader)
except SomeException as exc:
    text = str(exc)
except Exception:
    text = (
        "Switch asset failed. "
        "Search console log for more details."
    )

Reply:
We could - but in your code example, both exceptions would still need to result in the full pop-up dialog (duplicating the code). Right? That's why I wrote it this way to re-use the logic.

Also, I didn't inherit from the same errors because some were previously RuntimeErrors and others were ValueErrors and making them now inherit from the same errors would mean that any try..except someone might have used to capture the previous RuntimeErrors or ValueErrors might then not catch the new errors. So I've tried to inherit from the original errors where I could. (Except for the AssertionError that was there)

Reply:
> both exceptions would still need to result in the full pop-up dialog (duplicating the code).

It would just replace the `isintance` condition -> make more obvious different behavior.

> I didn't inherit from the same errors because some were previously RuntimeErrors and others were ValueErrors and making them now inherit from the same errors would mean that any try..except someone might have used to capture the previous RuntimeErrors or ValueErrors might then not catch the new errors.

To be honest I don't think we should keep that base exceptions. If anybody used `try: ... except RuntimeError:` in a custom logic (not sure where would someone used  custom logic for switching?) 1.) he should not do that 2.) he should know we added new exceptions.

This would work too.
```python
try:
    switch_container(container, repre_doc, loader)
except (
    LoaderSwitchNotImplementedError,
    IncompatibleLoaderError,
    LoaderNotFoundError,
) as exc:
    text = str(exc)
except Exception:
    text = (
        "Switch asset failed. "
        "Search console log for more details."
    )
```

Reply:
```python
try:
    switch_container(container, repre_doc, loader)
except (
    LoaderSwitchNotImplementedError,
    IncompatibleLoaderError,
    LoaderNotFoundError,
) as exc:
    text = str(exc)
except Exception:
    text = (
        "Switch asset failed. "
        "Search console log for more details."
    )
```

Yes, that would work. Except that instead it would become:
```python
try:
    switch_container(container, repre_doc, loader)
except (
    LoaderSwitchNotImplementedError,
    IncompatibleLoaderError,
    LoaderNotFoundError,
) as exc:
    text = str(exc)
        msg = (
        "Couldn't switch asset."
        "See traceback for more information."
    )
    log.warning(msg, exc_info=True)
    dialog = QtWidgets.QMessageBox(self)
    dialog.setWindowTitle("Switch asset failed")
    dialog.setText(text)
    dialog.exec_()
except Exception:
    text = (
        "Switch asset failed. "
        "Search console log for more details."
    )
    msg = (
        "Couldn't switch asset."
        "See traceback for more information."
    )
    log.warning(msg, exc_info=True)
    dialog = QtWidgets.QMessageBox(self)
    dialog.setWindowTitle("Switch asset failed")
    dialog.setText(text)
    dialog.exec_()
```
Right? Unless we move that logic into a re-useable function of its own?

Reply:
```
text = None
try:
    switch_container(container, repre_doc, loader)
except (
    LoaderSwitchNotImplementedError,
    IncompatibleLoaderError,
    LoaderNotFoundError,
) as exc:
    text = str(exc)
except Exception:
    text = (
        "Switch asset failed. "
        "Search console log for more details."
    )

if text:
    log.warning(msg, exc_info=True)
    dialog = QtWidgets.QMessageBox(self)
    dialog.setWindowTitle("Switch asset failed")
    dialog.setText(text)
    dialog.exec_()
```

Reply:
Fine with me - added with https://github.com/ynput/OpenPype/pull/5227/commits/6c96c658e26c9c7cded3e124abcccf7c3ebf507b